### ISSUE TITLE
DZP::ExtraTests should write dependency-free skip logic

### ISSUE THREAD

**Issue by kentfredric** [rjbs/Dist-Zilla#546]
As seen in https://rt.cpan.org/Ticket/Display.html?id=114237 , excessive application of DZP::ExtraTests wastes a lot of CPU time to basically print stuff to the screen.

Its annoying enough to waste CPU time to display it, but its even more annoying having such a large volume of "use Test::More" simply to print a "skipped" line and exit.

Seems this is a very good candidate for not loading Test::More.

**Comment by dolmen** [comment]
ExtraTests is an awful solution to a legitimate issue. We have now DZP::RunExtraTests (but not in dzil core) that does the job much better.
I suggested to @RJBS to deprecate ExtraTests in dzil v6.

**Comment by kentfredric** [comment]
It is an awful solution, indeed. But we can mitigate the overhead of using it slightly for those who insist on using it.

(Fwiw, I habitually nuke re-written tests in the vendor side testing chain because their existence just causes confusion and doesn't really help anyone)

**Comment by kentfredric** [comment]
Probably worth mentioning the tax of this approach is greatly amplified by Test2's increase of modules.

Its probably the worst-case for Test2.

**Comment by dolmen** [comment]
But Test2 also allows to change the formatter to something else than TAP which this would not allow:
```
BEGIN { unless ($ENV{RELEASE_TESTING}) { print "1..0 # SKIP these test are for release candidate testing\n"; exit } }
```

**Comment by kentfredric** [comment]
> But Test2 also allows to change the formatter to something else than TAP which this would not allow:

Quite, but given the proliferation of competing ways to emit tap that are used in the wild ( one of which is legacy Test.pm ), worrying "It could break because some harness might stop supporting TAP" seems like the worry is in the wrong place.

I'd be much more worried if Test::Harness decided it was a good time to start rejecting test-output emitters other than Test::More/Test2 

But then again...... I'm not sure what to expect anymore.

**Comment by dolmen** [comment]
I was only raising the Test2 formatter feature because the use case you linked in the initial report is a project from DROLSKY and [he clearly expressed his interest in using alternate formatters thanks to Test2](http://blogs.perl.org/users/chad_exodist_granum/2016/04/test2testbuilder-update-from-the-qah.html#comment-1702928).

**Comment by kentfredric** [comment]
Ah. Right. Fair point. 

So, in your estimation, would it make sense to support an option that dictated it would emit a "thin" rewrite instead of a "fat" rewrite?

Or do you imagine that people will expect to be able to change the formatter on code they didn't write, and making a "thin" version would be a negative thing for that goal?

**Comment by dolmen** [comment]
I don't think an option is necessary. I'm in favor of moving forward and emitting only the thin code (bypassing Test::More) that I wrote [above](#issuecomment-218498433).
If a dzil user in inconvenienced by this new behavior, using something else than ExtraTests remains possible.

**Comment by karenetheridge** [comment]
Agree. Keep it simple. The people who are using ExtraTests aren't the same
people who read the documentation and would use a config option, anway :)

**Comment by rjbs** [comment]
If you're not using the TAP formatter, you already need to be using a different harness.  I don't think we need to worry about this case.

**Comment by rjbs** [comment]
Merged #551.
